plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/statistics/FileBasedTableStatisticsProvider.java

```diff
@@ -102,7 +103,7 @@ public TableStatistics getTableStatistics(ConnectorSession session, DeltaLakeTab
         }
 
         Set<String> predicatedColumnNames = tableHandle.getNonPartitionConstraint().getDomains().orElseThrow().keySet().stream()
-                .map(DeltaLakeColumnHandle::getName)
+                .map(DeltaLakeColumnHandle::getBaseColumnName)
```

Comment:
Yes, `tableHandle.getNonPartitionConstraint()` will include the dereferenced field if used as a predicate. Currently, it is being treated as a base column. Since stats for the complex field are not supported, So stats is null.

should we use projected column name here if it dereference field?

I am purposing to add a test in `TestDeltaLakeTableStatistics`.

```java
    @Test
    public void testShowStatsForSelectNestedFieldWithWhereClause()
    {
        String tableName = "show_stats_select_nested_field_with_where_clause_" + randomNameSuffix();

        assertUpdate(
                "CREATE TABLE " + tableName + " (pk, int_col, row_col)" +
                        "WITH(partitioned_by = ARRAY['pk']) " +
                        "AS VALUES " +
                        "('pk1', null, CAST(ROW(23, 'field1') AS ROW(f1 INT, f2 VARCHAR))), " +
                        "(null, 12, CAST(ROW(24, 'field2') AS ROW(f1 INT, f2 VARCHAR))), " +
                        "('pk1', 13, CAST(ROW(25, null) AS ROW(f1 INT, f2 VARCHAR))), " +
                        "('pk1', 14, CAST(ROW(26, 'field1') AS ROW(f1 INT, f2 VARCHAR)))",
                4);
        assertQuery(
                "SHOW STATS FOR (SELECT int_col, row_col.f1, row_col FROM " + tableName + " WHERE row_col.f2 IS NOT NULL)",
                "VALUES " +
                        //  column_name | data_size | distinct_values_count | nulls_fraction | row_count | low_value | high_value
                        "('int_col', null, 3.0, 0.25, null, 12, 14), " +
                        "('f1', null, null, null, null, null, null), " +
                        "('row_col', null, null, null, null, null, null), " +
                        "(null, null, null, null, null, null, null)");
    }
```

Reply:
in retrospect, we should have left some code comment explaining why it's OK to call `getBaseColumnName` and not eg `getColumnName` or `verify(isBaseColumn())`.

please follow-up

Reply:
https://github.com/trinodb/trino/pull/17449